src/main/scala/scorex/api/http/TransactionsApiRoute.scala

```diff
@@ -56,21 +57,122 @@ case class TransactionsApiRoute(
               path(Segment) { limitStr =>
                 Exception.allCatch.opt(limitStr.toInt) match {
                   case Some(limit) if limit > 0 && limit <= MaxTransactionsPerRequest =>
-                    complete(Json.arr(JsArray(state.accountTransactions(a, limit).map{ case (h, tx) =>
+                    complete(Json.arr(JsArray(state.accountTransactions(a, limit, 0).map { case (h, tx) =>
                       processedTxToExtendedJson(tx) + ("height" -> JsNumber(h))
+                    })))
+                  case Some(limit) if limit > MaxTransactionsPerRequest =>
+                    complete(TooBigArrayAllocation)
+                  case _ =>
+                    complete(invalidLimit)
+                }
+              }
+          }
+      }
+    }
+  }
+
+  private val invalidOffset = StatusCodes.BadRequest -> Json.obj("message" -> "invalid.offset")
+
+  @Path("/address/{address}/limit/{limit}/offset/{offset}")
```

Comment:
we may combine this API with the previous one, and simplify the usage.

Reply:
OK, i'll change the path params to query params

Reply:
and it looks all these four API can be combined. ^^